dashboard/lti.py

```diff
@@ -27,13 +32,13 @@ def valid_lti_request(user_payload, request):
         login(request, user_obj)
     else:
         #handle no username from LTI launch
-        return HttpResponseRedirect(reverse('django_lti_auth:denied'))
+        raise LTIException("No username supplied in the launch, you should check your provider and/or settings.")
 
     url = reverse('home')
     if canvas_course_id:
         url = reverse('courses', kwargs={ 'course_id': canvas_course_id })
 
     return url
 
-# def invalid_lti_request(user_payload):
-#     pass
\ No newline at end of file
+def invalid_lti_request(user_payload):
+    logger.info(f"Invalid LTI request for {user_payload}")
```

Comment:
I think you need to add `HttpResponseRedirect(get_reverse('django_lti_auth:denied'))` in here if your using this function (default action taken by django-lti-auth https://github.com/rohitjose/django-lti-auth/blob/master/django_lti_auth/views.py#L56)

Reply:
Thanks, I'll call this. The LTI integration is looking really cool, happy how it's working for us. A few other schools have really been asking so I wanted to try it all out!

Reply:
It looks like from that link the code you're referencing will still run and return fine after the invalid_login_method_hook is executed. I tested and still getting the same "Sorry, you are not allowed to access this app" template from the module. That seems "fine" as you'd generally only see it if the key or secret are wrong.